llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp

```diff
@@ -1927,6 +1921,29 @@ void VPScalarCastRecipe ::print(raw_ostream &O, const Twine &Indent,
 }
 #endif
 
+void VPScalarStoreRecipe::execute(VPTransformState &State) {
+  [[maybe_unused]] bool IsAllOperandsUniformAfterVectorization =
```

Comment:
assert that both the operand and the recipe itself are defined outside the vector loop region, otherwise we might get unexpected results?